modules/zarr/src/lib/utils.ts

```diff
@@ -0,0 +1,106 @@
+import { openGroup } from 'zarr';
+import type { ZarrArray } from 'zarr';
+import type { PixelSource } from '../types';
+
+interface Multiscale {
+  datasets: { path: string }[];
+  version?: string;
+}
+
+export interface RootAttrs {
+  multiscales: Multiscale[];
+}
+
+export async function loadMultiscales(store: ZarrArray['store'], path = '') {
+  const grp = await openGroup(store, path);
+  const rootAttrs = (await grp.attrs.asObject()) as RootAttrs;
+
+  let paths = ['0'];
+  if ('multiscales' in rootAttrs) {
+    const { datasets } = rootAttrs.multiscales[0];
+    paths = datasets.map(d => d.path);
+  }
+
+  const data = paths.map(path => grp.getItem(path));
+  return {
+    data: (await Promise.all(data)) as ZarrArray[],
+    rootAttrs
+  };
+}
+
+function prevPowerOf2(x: number) {
+  return 2 ** Math.floor(Math.log2(x));
+}
+
+export function guessTileSize(arr: ZarrArray) {
+  const interleaved = isInterleaved(arr.shape);
+  const [yChunk, xChunk] = arr.chunks.slice(interleaved ? -3 : -2);
```

Comment:
I think a comment might be helpful here to explain `-3` and `-2`.

Reply:
Also I would've expected the chunk size to be part of the initial metadata in the `.zattrs` file?

Reply:
`chunks` is included in the `.zarray` metadata (Array metadata). https://zarr.readthedocs.io/en/stable/spec/v2.html#arrays

This metadata initializes the `ZarrArray` object and can be accessed from `ZarrArray.chunks`. Zarr.js handles parsing `.zattrs`/`.zarray`.